test/new-e2e/pkg/e2e/kubernetes_dump.go

```diff
@@ -0,0 +1,291 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/user"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/util/pointer"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	awsec2 "github.com/aws/aws-sdk-go-v2/service/ec2"
+	awsec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	awseks "github.com/aws/aws-sdk-go-v2/service/eks"
+	awsekstypes "github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	kubectlget "k8s.io/kubectl/pkg/cmd/get"
+	kubectlutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+func dumpKubernetesClusterState(ctx context.Context, name string) string {
+	cfg, err := awsconfig.LoadDefaultConfig(ctx)
+	if err != nil {
+		return "WARNING: Failed to dump cluster state: failed to load AWS config"
+	}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
would this fail on local kind?

Reply:
The dump would fail yes. But it would not stop the teardown, only writing a `warning` in the logs. I think that dumping the state on local provider is not that important since you can easily see the state of the cluster when working locally